src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -316,14 +317,18 @@ private async Task LoadKeyValuesRegisteredForRefresh(IDictionary<string, Configu
             {
                 string watchedKey = changeWatcher.Key;
                 string watchedLabel = changeWatcher.Label;
+                KeyLabel watchedKeyLabel = new KeyLabel(watchedKey, watchedLabel);
 
                 // Skip the loading for the key-value in case it has already been loaded
-                if (data.ContainsKey(watchedKey) && data[watchedKey].Label == watchedLabel.NormalizeNull())
+                if (data.TryGetValue(watchedKey, out ConfigurationSetting loadedKv)
+                    && loadedKv.Key == watchedKey
+                    && loadedKv.Label == watchedLabel.NormalizeNull())

```

Comment:
>watchedLabel.NormalizeNull() [](start = 41, length = 28)

I'm thinking we can move this into the `KeyLabel` class equals comparison and use that here to see if they are queal.